### PR TITLE
feat: Provide access to bundled libraries when in browser

### PR DESCRIPTION
**Pull request by vasco-santos**
This PR aims to update the spec, as a result of the [Issue 406](https://github.com/ipfs/js-ipfs-api/issues/406).

According to the following PRs:

- `js-ipfs`: [PR 1297](https://github.com/ipfs/js-ipfs/pull/1297)
- `js-ipfs-api`:  [PR 732](https://github.com/ipfs/js-ipfs-api/pull/732)

### CODE REVIEW

File: SPEC/TYPES.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+A set of data types are exposed directly from the IPFS instance under `ipfs.types`. That way you're not required to import/require the following.
+
+- [`ipfs.types.Buffer`](https://www.npmjs.com/package/buffer)
+- [`ipfs.types.PeerId`](https://github.com/libp2p/js-peer-id)
+- [`ipfs.types.PeerInfo`](https://github.com/libp2p/js-peer-info)
+- [`ipfs.types.multiaddr`](https://github.com/multiformats/js-multiaddr)
+- [`ipfs.types.multibase`](https://github.com/multiformats/multibase)
+- [`ipfs.types.multihash`](https://github.com/multiformats/js-multihash)
+- [`ipfs.types.CID`](https://github.com/ipld/js-cid)
+- [`ipfs.types.crypto`](https://github.com/libp2p/js-libp2p-crypto)
```
Comment (diasdavid) [inline SPEC/TYPES.md]: Not a type.

File: SPEC/TYPES.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+- [`ipfs.types.dagPB`](https://github.com/ipld/js-ipld-dag-pb)
+- [`ipfs.types.dagCBOR`](https://github.com/ipld/js-ipld-dag-cbor)
+- [`ipfs.types.isIPFS`](https://github.com/ipfs-shipyard/is-ipfs)
```
Comment (diasdavid) [inline SPEC/TYPES.md]: Not a type.

File: SPEC/TYPES.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+- [`ipfs.types.Buffer`](https://www.npmjs.com/package/buffer)
+- [`ipfs.types.PeerId`](https://github.com/libp2p/js-peer-id)
+- [`ipfs.types.PeerInfo`](https://github.com/libp2p/js-peer-info)
+- [`ipfs.types.multiaddr`](https://github.com/multiformats/js-multiaddr)
+- [`ipfs.types.multibase`](https://github.com/multiformats/multibase)
+- [`ipfs.types.multihash`](https://github.com/multiformats/js-multihash)
+- [`ipfs.types.CID`](https://github.com/ipld/js-cid)
+- [`ipfs.types.crypto`](https://github.com/libp2p/js-libp2p-crypto)
+- [`ipfs.types.dagPB`](https://github.com/ipld/js-ipld-dag-pb)
+- [`ipfs.types.dagCBOR`](https://github.com/ipld/js-ipld-dag-cbor)
```
Comment (diasdavid) [inline SPEC/TYPES.md]: perhaps scope these under IPLD and add all the ones that are supported today?

File: SPEC/TYPES.md (hunk 1 above)
Comment (vasco-santos) [reply SPEC/TYPES.md]: Agreed. Should I create `util` scope for encompassing this?

File: SPEC/TYPES.md (hunk 3 above)
Comment (vasco-santos) [reply SPEC/TYPES.md]: I agree with your idea. However, I would have to add `ipld/js-cid` too. As a consequence, a breaking change would be created in `js-ipfs` [core/index.js#L62](https://github.com/ipfs/js-ipfs/blob/master/src/core/index.js#L62). What do you think?

Regarding adding the other ones, if we decide for creating the specific scope, I think we may add them as well. So, you are referring to all the not archived JS modules at [ipld](https://github.com/ipld?utf8=%E2%9C%93&q=&type=&language=javascript)?

File: SPEC/TYPES.md (hunk 1 above)
Comment (diasdavid) [reply SPEC/TYPES.md]: Good idea!

File: SPEC/TYPES.md (hunk 3 above)
Comment (diasdavid) [reply SPEC/TYPES.md]: That's a valid point, thanks for thinking of that. Let's keep it as it is then.

**Comment by vasco-santos** [comment]
Done @diasdavid 😄 

I also updated the other PRs accordingly.

File: js/src/util.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+'use strict'
+
+const crypto = require('libp2p-crypto')
+const isIPFS = require('is-ipfs')
+
+const chai = require('chai')
+const dirtyChai = require('dirty-chai')
+const expect = chai.expect
+chai.use(dirtyChai)
+util
```
Comment (alanshaw) [inline js/src/util.js]: `undefined`?

File: js/src/types.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+const multibase = require('multibase')
+const multihash = require('multihashes')
+const CID = require('cids')
+
+const chai = require('chai')
+const dirtyChai = require('dirty-chai')
+const expect = chai.expect
+chai.use(dirtyChai)
+
+describe('.types', function () {
```
Comment (alanshaw) [inline js/src/types.js]: Are these runnable? There's no `module.exports = (common) => {` wrapper and no entry in `js/src/index.js` for them.

File: js/src/util.js (hunk 4 above)
Comment (vasco-santos) [reply js/src/util.js]: That util was supposed to be inside `describe`. How should I proceed @diasdavid ? A new PR fixing this typo?

File: js/src/types.js (hunk 5 above)
Comment (diasdavid) [reply js/src/types.js]: @vasco-santos did you open a new PR?

File: js/src/util.js (hunk 4 above)
Comment (diasdavid) [reply js/src/util.js]: @vasco-santos yes please.

File: js/src/types.js (hunk 5 above)
Comment (vasco-santos) [reply js/src/types.js]: It is created now 🙂

**Pull request merged**